conans/client/conan_api.py

```diff
@@ -326,6 +326,12 @@ def export_pkg(self, conanfile_path, name, channel, source_folder=None, build_fo
                                  "files" % (CONANINFO, BUILD_INFO))
 
         conanfile_path = _get_conanfile_path(conanfile_path, cwd, py=True)
+
+        if package_folder:
+            if build_folder or source_folder:
+                raise ConanException("--package-folder incompatible with --build-folder and --source-folder")
```

Comment:
I would raise the error without the arg name but the parameter name=> not --package-folder but "package_folder". I think it is enough for warn command line users too, if don't add another check in the command.py module.